suno/suno.py

```diff
@@ -186,6 +186,48 @@ def get_songs(self, prompt: str) -> dict:
                 break
         # keep the song info dict as old api
         return self.song_info_dict
+    
+    def get_songs_custom(self, prompt: str, genre: str, instrumental : str = False) -> dict:
+        url = f"{base_url}/api/generate/v2/"
+        self.session.headers["user-agent"] = ua.random
+        payload = {
+            "mv": "chirp-v3-0",
+            "prompt": "" if instrumental else prompt,
+            "tags": genre,
+            "make_instrumental": instrumental,
+        }
+        response = self.session.post(
+            url,
+            data=json.dumps(payload),
+            impersonate=browser_version,
+        )
+        if not response.ok:
+            print(response.text)
+            raise Exception(f"Error response {str(response)}")
+        response_body = response.json()
+        songs_meta_info = response_body["clips"]
+        request_ids = [i["id"] for i in songs_meta_info]
+        start_wait = time.time()
+        print("Waiting for results...")
+        sleep_time = 10
+        while True:
+            if int(time.time() - start_wait) > 600:
+                raise Exception("Request timeout")
+            # TODOs support all mp3 here
+            song_info = self._fetch_songs_metadata(request_ids)
+            # spider rule
+            if sleep_time > 2:
+                time.sleep(sleep_time)
+                sleep_time -= 2
+            else:
+                time.sleep(2)
+
+            if not song_info:
+                print(".", end="", flush=True)
+            else:
+                break
+        # keep the song info dict as old api
+        return self.song_info_dict
```

Comment:
most of the code are the same as `get_songs` maybe we can refactor it in the future but it is OK for the PR